src/resources/js/views/events-bar.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * Makes sure we have all the required levels on the Tribe Object
+ *
+ * @since TBD
+ *
+ * @type   {PlainObject}
+ */
+tribe.events = tribe.events || {};
+tribe.events.views = tribe.events.views || {};
+
+/**
+ * Configures Events Bar Object in the Global Tribe variable
+ *
+ * @since TBD
+ *
+ * @type   {PlainObject}
+ */
+tribe.events.views.eventsBar = {};
+
+/**
+ * Initializes in a Strict env the code that manages the Event Views
+ *
+ * @since TBD
+ *
+ * @param  {PlainObject} $   jQuery
+ * @param  {PlainObject} obj tribe.events.views.manager
+ *
+ * @return {void}
+ */
+( function( $, obj ) {
+	'use strict';
+	var $document = $( document );
+
+	/**
+	 * Selectors used for configuration and setup
+	 *
+	 * @since TBD
+	 *
+	 * @type {PlainObject}
+	 */
+	obj.selectors = {
+		input: '.tribe-common-form-control-text__input',
+		inputWrapper: '.tribe-common-form-control-text',
+		inputKeywordWrapper: '.tribe-common-c-search__input-control--keyword',
+		inputKeywordWrapperFocus: '.tribe-common-c-search__input-control--keyword-focus',
+		inputLocationWrapper: '.tribe-common-c-search__input-control--location',
+		inputLocationWrapperFocus: '.tribe-common-c-search__input-control--location-focus',
+	};
+
+	/**
+	 * Toggles input class if it has a value
+	 *
+	 * @since TBD
+	 *
+	 * @param {Event} event event object of click event
+	 *
+	 * @return {void}
+	 */
+	obj.setInputFocusClass = function( event ) {
+		var $input = event.data.target;
+
+		// Set the focus class if it has content.
+		$input
+			.closest( obj.selectors.inputWrapper )
+			.toggleClass( event.data.inputClassFocus, '' !== $input.val().trim() );
+	};
+
+	/**
+	 * Bind events for the keyword input of the events bar
+	 *
+	 * @since TBD
+	 *
+	 * @param {jQuery} $container jQuery object of container.
+	 *
+	 * @return {void}
+	 */
+	obj.bindEventsInputKeyword = function( $container ) {
+
+		// Bind event for the keyword input.
+		$container
+			.find( obj.selectors.inputKeywordWrapper )
+			.find( obj.selectors.input )
```

Comment:
Instead of finding the input wrapper, then finding the input, then doing the `closest()` to get the wrapper again, why don't we `.find( obj.selectors.inputKeywordWrapper )`, toggle class, then find the input and add the `on( 'change' )`? Saves us from running multiple dom queries.

Reply:
Because, as the modifier class is in the wrapper now, we need to go and check the input value. Even if we do it that way, we would have to go through the DOM in a different way, but I'm not sure if we could "economize" anything here.

The class changes are tied to what's in the input.

Reply:
Agreed, we should able to more efficiently target for what you need to do here. Whether that's being more extensive in your use of `data-js` and/or adding additional hooks with which to reduce the amount of dom queries needed. Optimizations on how you target and select elements add up : )